### PR TITLE
fix pollfd

### PR DESCRIPTION
**Pull request by vtjnash**
see comment description

fixes https://github.com/JuliaLang/IJulia.jl/issues/331

### CODE REVIEW

**Comment by yuyichao** [comment]
I can confirm that this fixes the IJulia kernel hang. Thanks!.

**Comment by vtjnash** [comment]
for a good read on this bug, see http://marc.info/?l=zeromq-dev&m=134082860723965&w=2

**Comment by vtjnash** [comment]
@tkelman if you're online, could you cancel the appveyor builds in https://github.com/JuliaLang/ZMQ.jl/pull/86 and https://ci.appveyor.com/project/tkelman/zmq-jl/build/1.0.86 so we don't need to wait for them to eventually time out

**Comment by stevengj** [comment]
So this is ready to merge?

**Comment by tkelman** [comment]
This got rebased out, but I still don't think we should bother allowing this to be installed on pre-release versions of 0.3.

**Comment by tkelman** [comment]
And if we're raising the minimum supported Julia version, will be best to bump minor version when this gets tagged.

File: src/ZMQ.jl [hunk 1]
```diff
@@ -82,11 +76,24 @@ end
 macro v3only(ex)
     version.major >= 3 ? esc(ex) : :nothing
 end
+macro v3only(ex)
```
Comment (tkelman) [inline src/ZMQ.jl]: defined twice?

**Comment by vtjnash** [comment]
this should be good to merge, whenever someone wants to hit the big green button

**Comment by yuyichao** [comment]
And you are still claiming Pre 0.3 support?

**Comment by tanmaykm** [comment]
Bump. Good to merge and tag, I suppose?

**Comment by yuyichao** [comment]
Merged.

@tkelman If you think it's better to claim only 0.3+ please feel free to change that before tagging.

**Comment by tkelman** [comment]
I'd need to hear a convincing reason to claim support of prereleases of 0.3, yeah.

**Pull request merged**